net/peer.go

```diff
@@ -804,6 +808,30 @@ func (p *Peer) AddP2PCollections(collections []string) error {
 		addedTopics = append(addedTopics, col)
 	}
 
+	// If adding the collection topics succeeds, we remove the collections' documents
+	// from the pubsub topics to avoid receiving duplicate events.
+	for _, col := range collections {
+		c, err := store.GetCollectionBySchemaID(p.ctx, col)
+		if err != nil {
+			return err
+		}
+		keyChan, err := c.GetAllDocKeys(p.ctx)
+		if err != nil {
+			return err
+		}
+		for key := range keyChan {
+			err := p.server.removePubSubTopic(key.Key.String())
```

Comment:
suggestion: This will not be rolled back if the txn commit fails, but it should be.  I do not see this as a new issue however (just expanded), as the `addedTopics` block also suffers from this.  Strongly suggest creating a ticket to roll this back properly on commit error (might already be one, I'm not sure though).

Same issue exists in RemoveP2PCollections

Reply:
Ticket opened 👍

Reply:
cheers :)